src/reference/asciidoc/aggregator.adoc

```diff
@@ -114,6 +114,19 @@ This method will be invoked for aggregating messages as follows:
 
 NOTE: In the interest of code simplicity, and promoting best practices such as low coupling, testability, etc., the preferred way of implementing the aggregation logic is through a POJO, and using the XML or annotation support for configuring it in the application.
 
+If the `MessageGroupProcessor` 's `process` method returns a collection, it must be a collection of `Messge<?>` s.
```

Comment:
>we really can return a Collection<Object> of payload,

Only if the MGP is a subclass of `AbstractAggregatingMessageGroupProcessor` which builds a message with the collection as the payload.

>If the `MessageGroupProcessor` 's `process`

`s/process/processMessageGroup/`

A "raw" MGP can only return a collection of messages.

See https://github.com/spring-projects/spring-integration/blob/master/spring-integration-core/src/main/java/org/springframework/integration/aggregator/AbstractCorrelatingMessageHandler.java#L645